File: lua/3_largest_prime_factor.lua
local number = 600851475143
local factor = 0

while number > factor do
    factor = (function(n)
        for i = 2, math.sqrt(n) do
            if n % i == 0 then
                return i
            end
        end
        return n
    end)(number)

    number = number / factor
end

print(factor)

File: ts/3_largest_prime_factor.ts
export {};

let number: number = 600851475143;
let factor: number = 0;

while (number > factor) {
    factor = ((n: number) => {
        for (let i = 2; Math.sqrt(n); i++) {
            if (n % i == 0) {
                return i;
            }
        }
        return n;
    })(number);

    number /= factor;
}

console.log(factor)